### ISSUE TITLE
TaskQueues API Please!

### ISSUE THREAD

**Issue by unitydynamics** [GoogleCloudPlatform/google-cloud-java#682]
Is it part of the roadmap to fold in the TaskQueue library?  This is what I see right now...

https://developers.google.com/api-client-library/java/apis/taskqueue/v1beta2

It would great if this was part of gcloud-java.  

Update: I noticed gcloud-python as pub/sub API support... which made me think... should I be using pub/sub instead of task queues for new development?

**Comment by aozarov** [comment]
@unitydynamics The TaskQueue API you are referring to is a REST API to the App Engine Task Queue.

Therefore, I think, will only work if your project is configured with an AE application and configuration of the queues is done via AE deployments.

In `gcloud-java` we started work on adding a Pub/Sub API support (in the pubsub-alpha branch), however though both are messaging APIs Publish/Subscribe and Queues have different semantics and behaviour.

We do plan to support a Google Cloud Task Queue (which is not tied to AE) when that would be available. See #22.

**Comment by unitydynamics** [comment]
Thanks for note.  I need to get a handle on how to view Google's milestones.  In my use case I like TaskQueues as a communication between my compute-engine and app-engine code.  So I suppose I should clarify my feature request... if TaskQueues come to `gcloud-java` I would like to interact with app-engine queues and have the seemless authentication that gcloud-java offers.  (and maybe a better API interface... one that eventually doesn't have "beta" in the name...)

**Comment by unitydynamics** [comment]
Not sure where best to put this... so you'll have to excuse me.  I ran into a nasty "bug" which had me pulling out my hair.  Turns out the "https://www.googleapis.com/auth/cloud-platform" scope does NOT include "https://www.googleapis.com/auth/taskqueue"   Is that intentional?  Seems misleading at best.

**Comment by ajkannan** [comment]
For Task Queues, you can use the scopes listed on the [Task Queue REST API docs page]( https://cloud.google.com/appengine/docs/java/taskqueue/rest/about_auth).  It's not immediately clear to me why the cloud-platform scope doesn't include Task Queue, but I will ask around for an answer and get back to you on that.

**Comment by kimsterv** [comment]
Hi, the Task Queue REST API doesn't include the cloud-platform scope because it hasn't been a popular feature request and the REST API is an alpha feature. It's unlikely we will ever add support for it because we will be adding IAM integration which will solve this problem in a better way.

It would be great to hear about your requirements as we're working on IAM integration including defining the roles. Feel free to ping me directly at klewandowski@google.com

-Kim 
Task Queues Product Manger

**Comment by unitydynamics** [comment]
First of all thank you for the responses!  Just checking in... are TaskQueues still on the roadmap for cloud-java?  The "alpha" API has been working great this year... although lately I do tend to get a lot more transient type of errors (see below).  Looked at Pub/Sub but TaskQueues have scheduling, which I need.  Best, Robert

com.google.api.client.googleapis.json.GoogleJsonResponseException: 500 Internal Server Error
{
  "code" : 500,
  "errors" : [ {
    "domain" : "global",
    "message" : "An internal error happened in the backend",
    "reason" : "internalError"
  } ],
  "message" : "An internal error happened in the backend"
}

**Comment by mziccard** [comment]
Hi @unitydynamics, TaskQueues are still on the roadmap but I don't have an ETA for you yet :( /cc @omaray

**Comment by hraban** [comment]
Hi all, thanks for looking at this. In the mean time, I have a question: do any of you use task queues from the GAE flexible environment, and if so: how? I can't really find a way to do this without resorting to the -compat base images, which is quite constraining.

I recently wrote this answer to a Stackoverflow question, but maybe I overlooked something?

http://stackoverflow.com/a/39897507/4359699

Thanks
